src/aks-preview/azext_aks_preview/decorator.py

```diff
@@ -1301,6 +1305,43 @@ def set_up_network_profile(self, mc: ManagedCluster) -> ManagedCluster:
         mc = super().set_up_network_profile(mc)
         network_profile = mc.network_profile
 
+        ip_families = self.context.raw_param.get('ip_families')
+
+        if ip_families:
+            network_profile.ip_families = ip_families.split(',')
+
+        pod_cidrs = self.context.raw_param.get('pod_cidrs')
+        if pod_cidrs:
+            network_profile.pod_cidrs = pod_cidrs.split(',')
+
+        service_cidrs = self.context.raw_param.get('service_cidrs')
+        if service_cidrs:
+            network_profile.service_cidrs = service_cidrs.split(',')
```

Comment:
I would appreciate it if you could write corresponding "getter" functions under class `AKSPreviewContext` for these new parameters. An example would be [`_get_aad_admin_group_object_ids`](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/acs/decorator.py#L3381) from azure-cli. You don't need to write the internal function (function name starts with `_`) since you do not have extra validations for these parameters. Please do not modify the value if it is obtained from the `mc` object. The detailed explanation for this is as follows

> In the update mode, the parameter value is not obtained from `mc` because we just need to use the value provided by the user to overwrite the value in `mc`.
> 
> But in the create mode, `mc` is completely blank, and we want to assemble the user's input into a complete `mc`. Considering that our parameters have a certain dependency or mutual exclusion relationship, or the parameters will be normalized before putting into `mc`, there are even some parameters that do not need user to provide explicitly, but can be dynamically completed by cli.
> 
> In order to ensure the consistency of the state and make the value of the parameter more clearly determined, we created the following principle in the create mode, that is, if a parameter has been placed in the `mc` object, then when trying to get the parameter value again, (e.g. cross-validation with other parameters) the value in `mc` shall prevail.

Also in this way, we can write unit test cases for these "getter" functions in file `test_decorator.py`.

Reply:
added getters. Although I would like to get your opinion on the behavior. I was not aware the SDK provided default values for pod_cidr and service_cidr. For new functionality we want customers to be able to just do `az aks create --pod-cidrs XYZ,ABC --service-cidrs FOO,BAR` where these do not match the defaults for pod-cidr and service-cidr. Therefore, I've made the change here to overwrite whatever is in pod_cidr and service_cidr with pod_cidrs[0] and service_cidrs[0] if it exists. This would be required as RP validates this anyway.